src/shaders/skybient.frag

```diff
@@ -0,0 +1,24 @@
+uniform vec3 bottomColor;
+uniform vec3 topColor;
+uniform float offset;
+uniform float exponent;
+
+varying vec3 vWorldPosition;
```

Comment:
`uniform` is a variable that gets passed to ALL fragments (pixels) of the screen at the same time. It's **uniform** across the whole program. Kinda like a global constant.

`varying` is a variable that gets passed between the fragment and vertex shaders. In this case, `vWorldPosition` gets passed from the vertex program which determines the position of the pixel (which is literally the whole scene for this case), and passes all those positional numbers into the fragment shader.

Because a fragment shader runs a parallel operation on every single pixel on the screen, I'm able to define a gradient across it by using the `vWorldPosition` as my input and outputting an rgb value depending on that value. Sorry if that's a bit confusing, but I hope it makes some sense. I'm just starting to learn this stuff as well.